locker

```diff
@@ -7,41 +7,72 @@ function lock_image() {
 	echo $NAME@$DIGEST
 }
 
-function lock_dockerfile() {
-	FILE=$1
+function unlock_image() {
+	IMAGE=$1
+	NAME=$(echo $IMAGE | grep '^[^@]\+' -o)
+	echo $NAME
+}
+
+function update_dockerfile() {
+	OPERATION=$1
+	FILE=$2
 	LOCKFILE=$FILE.lock
 
 	while IFS= read -r LINE; do
-		if [[ $LINE =~ ^FROM ]]
+		if [[ $LINE =~ ^FROM ]] && [[ $OPERATION == "lock" ]]
 		then
 			IMAGE=$(echo "$LINE" | sed -r 's/^FROM +([^ ]+).*$/\1/')
 			LOCKIMAGE=$(lock_image $IMAGE)
 			echo "$LINE" | sed -r "s|$IMAGE|$LOCKIMAGE|" >>$LOCKFILE
+		elif [[ $LINE =~ ^FROM ]] && [[ $OPERATION == "unlock" ]]
+		then
+			IMAGE=$(echo "$LINE" | sed -r 's/^FROM +([^ ]+).*$/\1/')
+			LOCKIMAGE=$(unlock_image $IMAGE)
+			echo "$LINE" | sed -r "s|$IMAGE|$LOCKIMAGE|" >>$LOCKFILE
 		else
 			echo "$LINE" >>$LOCKFILE
 		fi
 	done <$FILE
 
 	mv $LOCKFILE $FILE
-
-	echo "Locked $FILE"
 }
 
-if [ $# -ne 1 ]
+if [ $# -lt 1 ] || [ $# -gt 2 ]
 then
-	echo "Usage: locker FILE"
+	echo "Usage: locker FILE [-u | --unlock]"
 	exit 1
 fi
 
 FILE=$1
 FILEPATH=$(readlink -f $FILE)
 FILENAME=$(basename $FILEPATH)
 
-case $FILENAME in
-	"Dockerfile")
-		lock_dockerfile $FILEPATH
+case $2 in
+	--unlock | -u)
+		case $FILENAME in
+			"Dockerfile")
+				update_dockerfile "unlock" $FILEPATH
+				echo "Unlocked $FILE"
+				;;
+			*)
+				echo "Unknown file: $FILENAME"
+				;;
+		esac
+		;;
+	"")
+		case $FILENAME in
+			"Dockerfile")
+				update_dockerfile "lock" $FILEPATH
+				echo "Locked $FILE"
+				;;
+			*)
+				echo "Unknown file: $FILENAME"
+				;;
+		esac
```

Comment:
It'd be good to keep the top-level switch on filename but pass through the operation (lock/unlock) as an argument. That should stop the duplication.